### PR TITLE
Handle MIME parameters in Content-Type Header

### PR DESCRIPTION
**Pull request by sagar23sj**
* Handled MIME parameters in Content-Type Header
* Added test-cases to support the changes

Fixes: #1874

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Beautiful!  Thank you, @sagar23sj !
LGTM.

Awaiting second LGTM before merging.

**Review by Parker77 — APPROVED** [review]
LGTM.

**Pull request merged**
